### ISSUE TITLE
XAML Generator fails when Project Name includes a space 

### ISSUE THREAD

**Issue by dansiegel** [unoplatform/uno#14596]
### Current behavior

When creating a project with the latest template where the project name contains a space like “Contoso Dashboard” the XAML generator fails. 

### Expected behavior

The generator should work even though the assembly name and path contain spaces. 

### Workaround

I believe this fixed the issue
`<AssemblyName>$(MSBuildProjectName.Replace(‘ ‘, ‘_’))</AssemblyName>`

### Works on UWP/WinUI

None

**Comment by dansiegel** [comment]
See unoplatform/uno.templates#474

These issues seem to be coming from the Uno.UI.SourceGenerators. Adding `EmitCompilerGeneratedFiles` to view the generated source exposes some issues with inconsistency in how the genreators determine the namespace.

Creating a new project using the Uno.Templates with the name `With@Sign` the errors seem to be coming from the XamlCodeGenerator specifically. To make it more fun the logic to determine the appropriate namespace for the generated code seems to be different depending on what we generate.

- `AppResources_40783c46fac9e95f67d1a83dc9369301.cs` - has a `using With@Sign;` and `namepsace `With@Sign`
- `GlobalStaticResources.cs` - has namespace With@Sign`
- `MainPage_79e7eb5d7d2260e22c3d3dce8580957f.cs` - has `using With@Sign;` and `namespace With_Sign`

**Comment by Youssef1313** [comment]
Even WinUI seems to fail for this case.

![image](https://github.com/unoplatform/uno/assets/31348972/79094d47-f8d7-4304-b115-09b108e827af)

I think it might not be worth fixing if WinUI doesn't work.

@jeromelaban thoughts?

**Comment by Youssef1313** [comment]
This also might be fixable in the templates themselves by having a proper value for `RootNamespace`

**Comment by Youssef1313** [comment]
Note that it seems to work on my side when the name includes a space:

![image](https://github.com/unoplatform/uno/assets/31348972/0c77f0f8-22dc-423a-9651-9fa198b4d140)

It's only problematic for `@` which is the same behavior as in WinUI that I think having proper RootNamespace value would make it work.

**Comment by jeromelaban** [comment]
Right, if WinUI does not support it properly, I don't think it makes sense for now. We can avoid having invalid project names in the vsix, but we can't really do that in `dotnet new` (right, @dansiegel ?).

If we really can't, we could fail the build explicitly using an msbuild validation target.

**Comment by dansiegel** [comment]
While we have some work arounds that provide a more usable error message now, I would still say that our generators should be using the RootNamespace from MSBuild and not relying on the Assembly Name for the namespace during source generation.
